Lib/test/test_functools.py

```diff
@@ -1015,6 +1015,12 @@ class TestReduceC(TestReduce, unittest.TestCase):
 class TestReducePy(TestReduce, unittest.TestCase):
     reduce = staticmethod(py_functools.reduce)
 
+    def test_reduce_with_kwargs(self):
+        with self.assertWarns(DeprecationWarning):
+            self.reduce(function=lambda x, y: x + y, sequence=[1, 2, 3, 4, 5], initial=1)
```

Comment:
I think `sequence` kwarg should be removed here.

Reply:
Then it would follow the keyword argument, and raise a `SyntaxError`

Reply:
Yeah, positional argument should be moved.

Reply:
Sorry Sergey, but I don't understand how it should be moved. Could you please show me how it looks?

Reply:
`self.reduce([1, 2, 3, 4, 5], function=lambda x, y: x + y, initial=1)` :)  Maybe this case should be rather added.

Reply:
It would raise a `TypeError`:
```python
Traceback (most recent call last):
  File "/home/eclips4/programming/programming-languages/cpython/Lib/test/test_functools.py", line 1020, in test_reduce_with_kwargs
    self.reduce([1, 2, 3, 4, 5], function=lambda x, y: x + y, initial=1)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/eclips4/programming/programming-languages/cpython/Lib/functools.py", line 1134, in wrapper
    return func(*args, **kwargs)
TypeError: reduce() got multiple values for argument 'function'
```